standards/technical-approach.md

```diff
@@ -0,0 +1,29 @@
+# Technical Approach Documents
+
+Technical Approach Documents (TADs) serve to help the engineering team document solutions to feature development, share ideas between teams and help highlight potential issues with tasks before they are encountered during the development process.
+
+TADs should strive to describe the whats, hows and whys of the technical problem that is trying to be solved. It should strive to understand the problem at hand in a comprehensive way and explore any challenges so that engineers can be confident in their implementations of solutions. In short writing a TAD should answer any technical questions prior to the start of actual development work, freeing up developers to focus on implementation once the major questions have been resolved.
+
+## Process
+
+TADs do not have to be created for all development work, but MUST be created in the following cases:
+
+- If a Business Requirements Document (BRD) has been created
+- If the scope of work is expected to extend beyond a single sprint
+- If multiple developers will be contributing to the new feature
```

Comment:
Would it be helpful to add something like "If there are multiple solutions to fix a problem or implement a feature and the most appropriate approach should be selected based on specifications and requirements"?

Reply:
Also, "If the pros and cons of introducing a new tool or technology to a project, a team, or the department need to be considered".

Reply:
Agreed that these are both conditions where a TAD would help, but not sure how best to phrase the first one because that is generally true except for pretty basic bugfixes (and I don't want to make it seem like people have to generate these _all_ the time).

Ideally a dev would use their discretion to generate a TAD in the case when they are considering two solutions to document/explain their decision. Maybe something like "If multiple approaches to a feature or problem are valid and the developer wishes to document their approach given the specifications and requirements." (?)

Reply:
Yes, that phrasing is good but what if an approach isn't valid and the developer doesn't know it or is doubtful about it? Maybe they are also looking for validation on their approach and want to clear it through someone more senior.

Reply:
I've added these as options when a TAD `should` be created. Hopefully this gives a good sense of these kinds of situations? (and hopefully shows the TAD process as a positive one):

```
- If multiple approaches to a feature or problem are valid and the engineer assigned the ticket(s) wishes to document the decisions made
- If questions are raised about the appropriateness of an approach and engineer/team wishes to document the decision process
- If a engineer is faced with a decision or choice that they would like to receive feedback on and/or have validated by a more senior member of the engineering team
```